### PR TITLE
Update php_codesniffer and fix errors.

### PR DESCRIPTION
**Pull request by derweili**
## Summary

Addresses issue:

- #6231

## Relevant technical choices

- Update php_codesniffer.
- Fix codesniffer errors.
- Remove `phpcs:ignore` ignore comments.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "phpunit/phpunit": "^5 || ^6 || ^7 || ^8 || ^9",
     "roave/security-advisories": "dev-latest",
     "roots/wordpress": ">=5.2",
-    "squizlabs/php_codesniffer": "^3.3",
+    "squizlabs/php_codesniffer": "3.7.1",
```
Comment (tofumatt) [inline composer.json:14]:
```suggestion
    "squizlabs/php_codesniffer": "^3.7",
```

We tend to not use as fixed version numbers in `composer.json`, to allow for patch updates.

**Review by tofumatt — COMMENTED** [review]
Looks like this is failing PHP 8 unit tests, but I'm wondering if that's related to #6418, which I think was merged after this.

I'll merge in the latest `develop` to see if that fixes things, as this is otherwise good. Though we tend to not use as fixed version numbers in `composer.json` as supplied, allowing for patch updates, so I've proposed that as a change…

**Review by tofumatt — APPROVED** [review]
Merging in `develop` did the trick, looks good 👍🏻

File: composer.json (hunk 1 above)
Comment (kuasha420) [reply composer.json]: @tofumatt just a note that version constraint is slightly different in composer compared to npm. ie. `^3.7` will not only update to patch version, but also minor version (ie. 3.8...). For patch only, we need `^3.7.0`. This is an info I found out while working on this very IB. :smile: 

However all the deps in our composer JSON are in the former 2 segment style and allows (unintentionally?) for minor version update as well so this is not an issue specific to this one. 

Just wanted to flag. 

cc @aaemnnosttv @eugene-manuilov

**Pull request merged**